java/src/main/native/src/ColumnVectorJni.cpp

```diff
@@ -31,6 +31,48 @@
 #include "cudf_jni_apis.hpp"
 #include "dtype_utils.hpp"
 
+namespace cudf {
+namespace jni {
+  /**
+   * @brief Creates an empty column according to the type tree speicified
+   * in @p view
+   *
+   * An empty column contains zero elements and no validity mask.
+   *
+   * Unlike the 'cudf::make_empty_column', it takes care of the nested type by
+   * iterating the children columns in the @p view
+   *
+   * @param[in] view The input column view
+   * @return An empty column for the input column view
+   */
+  std::unique_ptr<cudf::column> make_empty_column(cudf::column_view& view) {
+    using ScalarType = cudf::scalar_type_t<cudf::size_type>;
+
+    cudf::type_id tid = view.type().id();
+    if (tid == cudf::type_id::LIST) {
+      if (view.num_children() != 2) {
+        throw jni_exception("List type requires two children(offset, data).");
+      }
+      // Only needs the second child.
+      auto data_view = view.child(1);
+      auto zero = cudf::make_numeric_scalar(cudf::data_type(cudf::type_id::INT32));
+      zero->set_valid(true);
+      static_cast<ScalarType *>(zero.get())->set_value(0);
```

Comment:
This is a quite expensive way to create an offset vector with a single 0 value in it. It would be much quicker to use an rmm::device_vector which initializes its values an then build the offset column from that zero-initialized buffer.

Reply:
Good suggestion, updated